### PR TITLE
GH-31

### PR DESCRIPTION
**Pull request by akila-sama**

### CODE REVIEW

File: classes/class-pluginpage.php [hunk 1]
```diff
@@ -57,19 +71,72 @@ public function ak_enqueue_submenu_ajax_script() {
 		}
 	}
 
+
+	/**
+	 * Add a custom menu page.
+	 */
+	public function ak_custom_menu() {
```
Comment (uvesh-bhatti) [inline classes/class-pluginpage.php:79]: please add proper doc block to all function
follow the doc block : [link](https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/)

File: classes/class-portfolio.php [hunk 2]
```diff
@@ -111,12 +135,14 @@ public function save_custom_fields( $post_id ) {
...
-	* @param array $columns Existing columns.
-	* @return array Modified columns.
-	*/
-	public function custom_portfolio_columns( $columns ) {
+	 * Customize portfolio columns.
+	 *
+	 * Adds custom columns to the 'portfolio' post type list table.
+	 *
+	 * @param array $columns Existing columns.
+	 * @return array Modified columns.
```
Comment (uvesh-bhatti) [inline classes/class-portfolio.php:139]: also add  @ since in doc block

**Comment by uvesh-bhatti** [comment]
- please recheck the Portfolio item details meta box it give the warnings.
- mail is not send to user.

**Comment by dharmendra-vadher** [comment]
image

File: akila-portfolio.php
Comment (dharmendra-vadher) [inline akila-portfolio.php:1]:
```
// Ensure the file is being called by WordPress
if ( ! defined( 'ABSPATH' ) ) {
    exit;
}
```

File: akila-portfolio.php [hunk 3]
```diff
@@ -7,18 +7,40 @@
...
+if ( ! defined( 'AKILA_PORTFOLIO_PLUGIN_DIR' ) ) {
+	define( 'AKILA_PORTFOLIO_PLUGIN_DIR', plugin_dir_path( __FILE__ ) );
+}
+
+require_once AKILA_PORTFOLIO_PLUGIN_DIR . 'classes/class-portfolio.php';
+require_once AKILA_PORTFOLIO_PLUGIN_DIR . 'classes/class-pluginpage.php';
+require_once AKILA_PORTFOLIO_PLUGIN_DIR . 'classes/class-shortcodes.php';
+require_once AKILA_PORTFOLIO_PLUGIN_DIR . 'classes/class-button.php';
+require_once AKILA_PORTFOLIO_PLUGIN_DIR . 'classes/class-endpoints.php';
+require_once AKILA_PORTFOLIO_PLUGIN_DIR . 'classes/class-cron.php'; // Include the Cron class
```
Comment (dharmendra-vadher) [inline akila-portfolio.php]: Remove extra comments

File: akila-portfolio.php [hunk 4]
```diff
@@ -7,18 +7,40 @@
...
+
+new Akila\Portfolio\Portfolio(); // Initializes the Portfolio class for handling portfolio items
+new Akila\Portfolio\PluginPage(); // Initializes the PluginPage class for admin page enhancements
+new Akila\Portfolio\Shortcodes(); // Initializes the Shortcodes class for managing shortcodes
+new Akila\Portfolio\Button(); // Initializes the Button class for adding a custom button in the plugins page
+new Akila\Portfolio\Endpoints(); // Initializes the Endpoints class for custom REST API endpoints
```
Comment (dharmendra-vadher) [inline akila-portfolio.php]: Namespace should represent directory structure for better readability. 
`namespace APortfolio\Classes;` represent better namespace than `namespace Akila\Portfolio;`

File: classes/class-portfolio.php [hunk 5]
```diff
@@ -2,22 +2,40 @@
 
 namespace Akila\Portfolio;
 
+if ( ! defined( 'AKILA_PORTFOLIO_PLUGIN_DIR' ) ) {
+	define( 'AKILA_PORTFOLIO_PLUGIN_DIR', plugin_dir_path( __FILE__ ) );
+}
```
Comment (dharmendra-vadher) [inline classes/class-portfolio.php]: Isn't this declared in the main file.

File: akila-portfolio.php [hunk 6]
```diff
@@ -7,18 +7,40 @@
...
 
+// Activation hook
+function cpp_activate_plugin() {
```
Comment (dharmendra-vadher) [inline akila-portfolio.php]: Ensure consistent use prefix.

File: akila-portfolio.php [hunk 7]
```diff
@@ -7,18 +7,40 @@
...
-new Akila\Portfolio\Button();
-new Akila\Portfolio\Endpoints();
+// Deactivation hook
+function cpp_deactivate_plugin() {
+	// Change permalink structure to Plain
+	global $wp_rewrite;
+	$wp_rewrite->set_permalink_structure( '' );
+	$wp_rewrite->flush_rules(); // To make sure the changes take effect immediately
+}
+register_deactivation_hook( __FILE__, 'cpp_deactivate_plugin' );
```
Comment (dharmendra-vadher) [inline akila-portfolio.php]: Please add proper comment and remove extra comment.

File: classes/class-cron.php
Comment (dharmendra-vadher) [inline classes/class-cron.php:1]: Add documentaitons.
Activation deactivation hooks should be in the main file.

File: classes/class-cron.php [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+		// Loop through portfolio posts
+		foreach ( $portfolio_posts as $post_id ) {
+			// Get email address from post meta
+			$email = get_post_meta( $post_id, 'email', true );
+
+			// Check if email exists and send email
+			if ( ! empty( $email ) ) {
+				$subject = 'Portfolio Notification';
+				$message = 'This is a daily notification for your portfolio post.';
```
Comment (dharmendra-vadher) [inline classes/class-cron.php]: Add esc and translation functions

File: templates/plugin-details.php
Comment (dharmendra-vadher) [inline templates/plugin-details.php:1]: Please add proper docblock to all the files

**Pull request merged**